libyul/optimiser/LoadResolver.cpp

```diff
@@ -76,3 +105,62 @@ void LoadResolver::tryResolve(
 			if (inScope(*value))
 				_e = Identifier{locationOf(_e), *value};
 }
+
+void LoadResolver::tryEvaluateKeccak(
+	Expression& _e,
+	std::vector<Expression> const& _arguments
+)
+{
+	// Runtime saving: [PUSH32 32-byte-value] instead of [PUSH1 0 PUSH1 32 KECCAK256]
```

Comment:
Can you use GasMeter in EVMMetrics.h instead? Maybe you can also add a function `literalGas(size_t _bytes)` that you then call from `GasMeterVisitor::operator()(Literal const& _lit)`?